libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/route/RouteEngine.java

```diff
@@ -35,15 +36,15 @@ public void fetchRoute(Point origin, RouteProgress routeProgress) {
     this.routeProgress = routeProgress;
 
     // Calculate remaining waypoints
-    List<Point> coordinates = new ArrayList<>(routeProgress.directionsRoute().routeOptions().coordinates());
+    List<Point> waypoints = RouteUtils.calculateRemainingWaypoints(routeProgress);
 
-    if (coordinates.size() < routeProgress.remainingWaypoints()) {
+    if (waypoints == null) {
+      Timber.e("An error occurred fetching a new route");
       return;
     }
-    // Remove any waypoints that have been passed
-    coordinates.subList(0, routeProgress.remainingWaypoints()).clear();
+
     // Get the destination waypoint (last in the list)
```

Comment:
Could we extract `waypoints.size() - 1` into an explaining variable called `last` and remove the comment?

Reply:
Also it's kinda confusing that calling `remove` returns the element removed so maybe it's preferable to extract `waypoints.remove(waypoints.size() - 1)` into a method reflecting that. What do you think?